### ISSUE TITLE
The game crashes after an attack on the city

### ISSUE THREAD

**Issue by Povelitelh3** [vcmi/vcmi#2890]
Sorry if it's a duplicate. The bottom line is this: the teal AI attacks the city and the game crashes.

Giving a message like:

```
[2023-Sep-21 14:46:42.468627] DEBUG ai - Player 6 (teal): I won the Starting battle of Mrxxklhk attacking Base 44trd17, Conflux at (3 142 1)!
[2023-Sep-21 14:46:42.468906] DEBUG ai - Adding query 503 - Combat result dialog. Total queries count: 1
[2023-Sep-21 14:46:42.469462] TRACE network - 	Made first apply on cl: 12BattleResult
[2023-Sep-21 14:46:42.469504] DEBUG ai - I'll answer the query 503 giving the choice 0
[2023-Sep-21 14:46:42.470008] TRACE network - Sending a request "10QueryReply". It'll have an ID=622.
[2023-Sep-21 14:46:42.470284] TRACE network - Sending a pack of type 10QueryReply
[2023-Sep-21 14:46:42.470619] DEBUG ai - Attempted answering query 503 - Combat result dialog. Request id=622. Waiting for results...
[2023-Sep-21 14:46:42.470897] TRACE global - We'll wait till request 622 is answered.

[2023-Sep-21 14:46:42.471195] TRACE network - 	Applied on gs: 12BattleResult
[2023-Sep-21 14:46:42.471471] TRACE network - 	Made second apply on cl: 12BattleResult
[2023-Sep-21 14:46:42.472084] TRACE network - Received CPack of type 14SetStackEffect
[2023-Sep-21 14:46:42.472359] TRACE network - 	Made first apply on cl: 14SetStackEffect
[2023-Sep-21 14:46:42.472844] ERROR global - Disaster happened.
```
[atl.zip](https://github.com/vcmi/vcmi/files/12684094/atl.zip)
[logs.zip](https://github.com/vcmi/vcmi/files/12684095/logs.zip)

1.3.2
Windows 10

**Comment by Povelitelh3** [comment]
There was a second case on a random map. The heroes fought on the street. Could not reproduce. Only logs

```
[2023-Sep-21 21:27:30.741082] TRACE network - 	Applied on gs: 12BattleResult
[2023-Sep-21 21:27:30.741372] TRACE network - 	Made second apply on cl: 12BattleResult
[2023-Sep-21 21:27:30.741863] TRACE network - Received CPack of type 11StartAction
[2023-Sep-21 21:27:30.742133] ERROR global - Attempt to call battle interface without ongoing battle!
[2023-Sep-21 21:27:30.742389] TRACE network - 	Made first apply on cl: 11StartAction
[2023-Sep-21 21:27:30.742866] ERROR global - Disaster happened.
```

**Comment by IvanSavenko** [comment]
Fixed in 1.4
